task1.cpp

```diff
@@ -0,0 +1,195 @@
+	//StringListPlot2(list1);
+	
+	StringListRemoveDuplicates(list1);
+	StringListPlot(list1);
+	
+	StringListRemove(list1, str2);
+	StringListPlot(list1);
+
+	StringListRemove(list1, str11);
+	StringListPlot(list1);
+
+	StringListRemove(list1, str3);
+	StringListPlot(list1);
+	
+	StringListRemove(list1, str2);
+	StringListPlot(list1);
+
+	StringListRemove(list1, str11);
+	StringListPlot(list1);
+
+	StringListRemove(list1, str3);
+	StringListPlot(list1);
+
+	StringListRemoveDuplicates(list1);
+	StringListPlot(list1);
+
+	printf ("FIRST INDEX IS %i OF %s", StringListIndexOf(list1, str11), str11);
+	printf ("FIRST INDEX IS %i OF %s", StringListIndexOf(list1, str12), str12);
+	printf ("FIRST INDEX IS %i OF %s", StringListIndexOf(list1, str3), str3);
+
+	printf ("SIZE:%i\n", StringListSize(list1));
+	StringListDestroy(&list1);
+	printf ("list1:%p\n", list1);
+
+	free(str1);	free(str2);	free(str3);	free(str11);	free(str12);
+	return 0;
+}
+
+void StringListInit(char*** list){
+	if ( !(* list = (char**)malloc (2 * sizeof(char**))) )
+		return;
+	** list = 0;					
+	*(++(* list)) = 0;				
+}
+void StringListDestroy(char*** list) {/* Destroy list and set pointer to NULL. */
+	--(*list);
+	while (**list != 0) {
+		char ** nextItem = (char **)**list;
+		++nextItem;
+		nextItem = (char **)* nextItem;
+		free(**list);
+		** list = (char *)nextItem;
+	}
+	free(*list);
+	*list = 0;
+}
+
+void StringListAdd(char** list, char * str) {/* Inserts value at the end of the list. */
+	char ** item;
+	if ( !(item = (char**)malloc (2 * sizeof(char**))) )
+		return;
+	* item = str;					
+
+	if (*list == 0) {				//	first adding
+		* list = (char *)item;			
+		--list;
+		* list = (char *)item;			
+		*(++item) = 0; 				
+	} else {					//	next addings
+		char** last = (char**)*list;		
+		++last;					
+		* last = (char *)item;			
+		* list = (char *)item;			
+		*(++item) = 0;				
+	}
+}
+
+void StringListPlot(char** list) {
+	--list;
+	while (*list != 0) {
+		list = (char**)*list;
+		++list;
+	}
+}
+void StringListRemove(char** list, char* str) {/* Removes all occurrences of str in the list. */			
+	--list;
+	while (*list != 0) {
+		char ** nextItem = (char **)*list;				
+		if (!strcmp(*nextItem, str)){			
+			++nextItem;
+			nextItem = (char **)* nextItem;		
+			free(*list);			
+			*list = (char *)nextItem;	
+		} else {					
+			list = ++nextItem;
+		}
+	}
+}
+
+void StringListRemoveDuplicates(char** list) {/* Removes duplicate entries from the list. */	
+	--list;
+	while (*list != 0) {
+		char ** nextItem = (char **)*list;
+		char * str = *nextItem;	
+		StringListRemove(++++nextItem, str);
+		--nextItem;
+		list = (char **)nextItem;	
+	}
+}
+
+int StringListSize(char** list) {/* Returns the number of items in the list. */
+	--list;
+	int size = 0;
+	while (*list != 0) {
+		list = (char**)*list;
+		++size;
+		++list;
+	}
+	return size;
+}
+
+int StringListIndexOf(char** list, char* str) {/* Returns the index position of the first occurrence of str in the list. */
```

Comment:
Check if the list is not null and the string is not null. Try to use reinterpret_cast<Type>() instead of braces because it is hard to figure out that you are changing interpretation of the same data.